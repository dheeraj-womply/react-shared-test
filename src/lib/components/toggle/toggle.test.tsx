import React from "react";
import { create as renderer } from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react";
import { Toggle } from "./index";

const tabs = [
  { id: "0", label: "SMS", content: <div>SMS Tab Content</div> },
  { id: "1", label: "Email", content: <div>Email Tab Content</div> },
  { id: "2", label: "Slack", content: <div>Slack Tab Content</div> },
];

describe("Slider Tabs Matches the snapshots", () => {
  it("renders correctly with single tab", () => {
    const component = renderer(<Toggle tabs={tabs.slice(0, 1)} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with single tab and children", () => {
    const component = renderer(
      <Toggle tabs={[{ id: "0", label: "SMS" }]}>
        <div>Children</div>
      </Toggle>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with multiple tab", () => {
    const component = renderer(<Toggle tabs={tabs.slice(0, 2)} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with multiple tab and other selected tab", () => {
    const component = renderer(
      <Toggle defaultSelected={"1"} tabs={tabs} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders nothing with no data", () => {
    const component = renderer(<Toggle tabs={[]} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Slider Tabs component functionality is working", () => {
  beforeEach(() => {
    globalThis.HTMLElement.prototype.scrollBy = jest.fn();
    globalThis.HTMLElement.prototype.scrollTo = jest.fn();
  });

  it("renders correctly with no data", () => {
    const { container } = render(<Toggle tabs={[]} />);
    expect(container.innerHTML).toHaveLength(0);
  });

  it("renders correctly with default selected tab", () => {
    const { queryByTestId, getByTestId } = render(<Toggle hideArrows tabs={tabs} />);
    const sliderTabsContent = queryByTestId("slider-tabs-content");
    expect(sliderTabsContent).toHaveTextContent("SMS Tab Content");
    const input1 = getByTestId("input-1");
    fireEvent.click(input1);
    expect(queryByTestId("left-arrow")).toBeFalsy();
    expect(queryByTestId("right-arrow")).toBeFalsy();
  });

  it("renders correctly with data", () => {
    Object.defineProperty(HTMLElement.prototype, 'clientWidth', {
      configurable: true, value: 943
    });
    Object.defineProperty(HTMLElement.prototype, 'scrollWidth', {
      configurable: true, value: 1099
    });
    Object.defineProperty(HTMLElement.prototype, 'scrollLeft', {
      configurable: true, value: 10
    });
    const onTabChange = jest.fn();
    const { getByTestId } = render(
      <Toggle
        onTabChange={onTabChange}
        defaultSelected={"1"}
        tabs={Array.from({ length: 25 }, (v, i) => ({
          id: `${i}`,
          label: `Tab ${i}`,
          content: <div>Tab {i} Content</div>,
        }))}
      />
    );
    const sliderTabsContent = getByTestId("slider-tabs-content");
    expect(sliderTabsContent).toHaveTextContent("Tab 1 Content");
    const input0 = getByTestId("input-0");
    fireEvent.click(input0);
    expect(onTabChange).toBeCalled();
    expect(sliderTabsContent).toHaveTextContent("Tab 0 Content");
    const rightArrow = getByTestId("right-arrow");
    fireEvent.click(rightArrow);
    const leftArrow = getByTestId("left-arrow");
    fireEvent.click(leftArrow);
  });

  it("renders correctly with data and ref as null", () => {
    const { getByTestId } = render(
      <Toggle
        defaultSelected={"1"}
        tabs={Array.from({ length: 25 }, (v, i) => ({
          id: `${i}`,
          label: `Tab ${i}`,
          content: <div>Tab {i} Content</div>,
        }))}
      />
    );
    jest.spyOn(React, "useRef").mockReturnValue({ current: null });
    const input0 = getByTestId("input-0");
    fireEvent.click(input0);
  });
});
