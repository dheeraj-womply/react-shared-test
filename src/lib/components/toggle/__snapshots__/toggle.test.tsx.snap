// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Tabs Matches the snapshots renders correctly with multiple tab 1`] = `
<div
  className="sc-AxjAm gbNwSh"
>
  <div
    className="sc-AxirZ kkPxXZ"
    onScroll={[Function]}
  >
    <div
      className="sc-AxiKw dqSzmC"
    >
      <input
        data-testid="input-0"
        defaultChecked={true}
        id="0"
        name="tabs"
        onClick={[Function]}
        type="radio"
      />
      <label
        className="sc-AxhCb fNUnNf"
        htmlFor="0"
      >
        <span
          className="sc-AxheI hTfIJe"
        >
          SMS
        </span>
      </label>
    </div>
    <div
      className="sc-AxiKw dqSzmC"
    >
      <input
        data-testid="input-1"
        defaultChecked={false}
        id="1"
        name="tabs"
        onClick={[Function]}
        type="radio"
      />
      <label
        className="sc-AxhCb fNUnNf"
        htmlFor="1"
      >
        <span
          className="sc-AxheI hTfIJe"
        >
          Email
        </span>
      </label>
    </div>
    <span
      className="sc-AxhUy hUEDRh"
    />
  </div>
  <div
    data-testid="slider-tabs-content"
  >
    <div>
      SMS Tab Content
    </div>
  </div>
</div>
`;

exports[`Slider Tabs Matches the snapshots renders correctly with multiple tab and other selected tab 1`] = `
<div
  className="sc-AxjAm gbNwSh"
>
  <div
    className="sc-AxirZ kkPxXZ"
    onScroll={[Function]}
  >
    <div
      className="sc-AxiKw dqSzmC"
    >
      <input
        data-testid="input-0"
        defaultChecked={false}
        id="0"
        name="tabs"
        onClick={[Function]}
        type="radio"
      />
      <label
        className="sc-AxhCb fNUnNf"
        htmlFor="0"
      >
        <span
          className="sc-AxheI hTfIJe"
        >
          SMS
        </span>
      </label>
    </div>
    <div
      className="sc-AxiKw dqSzmC"
    >
      <input
        data-testid="input-1"
        defaultChecked={true}
        id="1"
        name="tabs"
        onClick={[Function]}
        type="radio"
      />
      <label
        className="sc-AxhCb fNUnNf"
        htmlFor="1"
      >
        <span
          className="sc-AxheI hTfIJe"
        >
          Email
        </span>
      </label>
    </div>
    <div
      className="sc-AxiKw dqSzmC"
    >
      <input
        data-testid="input-2"
        defaultChecked={false}
        id="2"
        name="tabs"
        onClick={[Function]}
        type="radio"
      />
      <label
        className="sc-AxhCb fNUnNf"
        htmlFor="2"
      >
        <span
          className="sc-AxheI hTfIJe"
        >
          Slack
        </span>
      </label>
    </div>
    <span
      className="sc-AxhUy hUEDRh"
    />
  </div>
  <div
    data-testid="slider-tabs-content"
  >
    <div>
      Email Tab Content
    </div>
  </div>
</div>
`;

exports[`Slider Tabs Matches the snapshots renders correctly with single tab 1`] = `
<div
  className="sc-AxjAm gbNwSh"
>
  <div
    className="sc-AxirZ kkPxXZ"
    onScroll={[Function]}
  >
    <div
      className="sc-AxiKw dqSzmC"
    >
      <input
        data-testid="input-0"
        defaultChecked={true}
        id="0"
        name="tabs"
        onClick={[Function]}
        type="radio"
      />
      <label
        className="sc-AxhCb fNUnNf"
        htmlFor="0"
      >
        <span
          className="sc-AxheI hTfIJe"
        >
          SMS
        </span>
      </label>
    </div>
    <span
      className="sc-AxhUy hUEDRh"
    />
  </div>
  <div
    data-testid="slider-tabs-content"
  >
    <div>
      SMS Tab Content
    </div>
  </div>
</div>
`;

exports[`Slider Tabs Matches the snapshots renders correctly with single tab and children 1`] = `
<div
  className="sc-AxjAm gbNwSh"
>
  <div
    className="sc-AxirZ kkPxXZ"
    onScroll={[Function]}
  >
    <div
      className="sc-AxiKw dqSzmC"
    >
      <input
        data-testid="input-0"
        defaultChecked={true}
        id="0"
        name="tabs"
        onClick={[Function]}
        type="radio"
      />
      <label
        className="sc-AxhCb fNUnNf"
        htmlFor="0"
      >
        <span
          className="sc-AxheI hTfIJe"
        >
          SMS
        </span>
      </label>
    </div>
    <span
      className="sc-AxhUy hUEDRh"
    />
  </div>
  <div
    data-testid="slider-tabs-content"
  >
    <div>
      Children
    </div>
  </div>
</div>
`;

exports[`Slider Tabs Matches the snapshots renders nothing with no data 1`] = `null`;
